packages/data-stores/src/wpcom-plans-ui/reducer.ts

```diff
@@ -25,9 +26,31 @@ const selectedStorageOptionForPlans: Reducer<
 	return state;
 };
 
+const comparisonGridCompactViewPlanDisplayOrder: Reducer<
+	PlanSlug[] | undefined,
+	WpcomPlansUIAction
+> = ( state, action ) => {
+	switch ( action.type ) {
+		case 'WPCOM_PLANS_UI_RESET_STORE':
+			return [];
+		case 'WPCOM_PLANS_UI_SET_COMPARISON_GRID_COMPACT_VIEW_PLAN_DISPLAY_ORDER':
+			return action.planOrder;
+		case 'WPCOM_PLANS_UI_COMPARISON_GRID_COMPACT_VIEW_SWITCH_PLAN': {
+			const currentPlanOrder = state ?? [];
+			return currentPlanOrder.map( ( planSlug ) =>
+				planSlug === action.currentPlan ? action.newPlan : planSlug
+			);
+		}
```

Comment:
Just to clarify - it isn't necessary to push this operation into the store, right? This could just happen at component level like before and simply store the "order" here.